server/src/main/java/org/opensearch/cluster/awarenesshealth/NodeShardInfo.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.awarenesshealth;
+
+/**
+ * Stores shard level information for a node
+ *
+ */
+public class NodeShardInfo {
+
+    private String nodeID;
+    private int activeShards;
+    private int relocatingShards;
+    private int initializingShards;
+
+    /**
+     * Stores shard level information for a node
+     *
+     * @param nodeID node id in the cluster
+     * @param activeShards active shards present on the node
+     * @param relocatingShards relocating shards present on the node
+     * @param initializingShards initialising shards present on the node
+     */
+    public NodeShardInfo(String nodeID, int activeShards, int relocatingShards, int initializingShards) {
```

Comment:
Lets have a constructor with just `nodeID` and functions to increment all the members instead .